service/status-report/main.go

```diff
@@ -77,6 +85,43 @@ func main() {
 		klog.Exitf("Failed to get storageClient %q/%q: %v", storageClient.Namespace, storageClient.Name, err)
 	}
 
+	csvList := opv1a1.ClusterServiceVersionList{}
+	if err = cl.List(ctx, &csvList, client.InNamespace(storageClientNamespace)); err != nil {
+		klog.Exitf("Failed to list csv resources: %v", err)
+	}
+
+	var opVersion string
+	for idx := range csvList.Items {
+		candidate := &csvList.Items[idx]
+		if strings.HasPrefix(candidate.Name, csvPrefix) {
+			opVersion = candidate.Spec.Version.String()
+			break
+		}
+	}
+
+	if opVersion == "" {
+		klog.Exitf("Unable to find csv with prefix %q", csvPrefix)
+	}
+
+	clusterVersion := &configv1.ClusterVersion{}
+	clusterVersion.Name = "version"
+	if err = cl.Get(ctx, types.NamespacedName{Name: clusterVersion.Name}, clusterVersion); err != nil {
+		klog.Exitf("Failed to get clusterVersion: %v", err)
```

Comment:
I am not sure that existing is a good result of not finding a piece of data. The reason is that this will basically indicate to the server that there is no connection from the client (because of the lack of a heartbeat) which is not true. 
It is always preferable to send what you have and let the server decide how to react

Reply:
done.